core/dbt/include/global_project/macros/adapters/timestamps.sql

```diff
@@ -0,0 +1,43 @@
+{%- macro current_timestamp() -%}
+    {{ adapter.dispatch('current_timestamp', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp() %}
+    current_timestamp
+{% endmacro %}
```

Comment:
`current_timestamp` as the default SQL for `{{ dbt.current_timestamp() }}` makes a lot of sense since it *is* a SQL standard 🧠 

But a few questions:
1. Do we _need_ a new default implementation here?
1. Do we _want_ a new default implementation here?
1. Why didn't the original author(s) do this in the first place?

### Context

This would replace the current implementation of `default__current_timestamp()` which is:

```sql
{% macro default__current_timestamp() -%}
  {{ exceptions.raise_not_implemented(
    'current_timestamp macro not implemented for adapter '+adapter.type()) }}
{%- endmacro %}
```

Tracing this implementation back in time goes to [here](https://github.com/dbt-labs/dbt-core/commit/5e8ab9ce4a8ac905d66bf85414a0be121fc58dbd#diff-29395c37ca37188832ed290a098706ca9b366055fa578d2524fde72ad6e3758dR233-R236).

Is this a [Chesterton's fence](https://sketchplanations.com/chestertons-fence)? I don't know! 🤷 We should figure it out one way or the other.

### Recommendation

If @jtcohen6 gives this new default implementation a 👍, I don't know of any technical reasons why _not_ to do it.

In a vacuum, my recommendation would be to leave the current implementation as-is _unless_ we can explain why the proposed replacement is more relevant than the previous implementation.

Reply:
[Quoting Chesterton](https://www.goodreads.com/quotes/833466-in-the-matter-of-reforming-things-as-distinct-from-deforming
):
> There exists in such a case a certain institution or law; let us say, for the sake of simplicity, a fence or gate erected across a road. The more modern type of reformer goes gaily up to it and says, “I don’t see the use of this; let us clear it away.” To which the more intelligent type of reformer will do well to answer: “If you don’t see the use of it, I certainly won’t let you clear it away. Go away and think. Then, when you can come back and tell me that you do see the use of it, I may allow you to destroy it.”

It isn't saying that something can't or shouldn't be reformed rather _how_ best to approach reforms.